common/src/main/java/org/hawkular/accounts/common/TokenVerifier.java

```diff
@@ -32,11 +39,41 @@
     @Inject @RealmName
     private String realm;
 
+    @Inject @HostSynonyms
+    private Set<String> hostSynonyms;
+
     @Inject
     AuthServerRequestExecutor executor;
 
     public String verify(String token) throws Exception {
-        String tokenUrl = baseUrl
+        JWSInput jwsInput;
+        try {
+            jwsInput = new JWSInput(token);
+        } catch (Exception e) {
+            throw new VerificationException("Couldn't parse token", e);
+        }
+
+        AccessToken accessToken;
+        try {
+            accessToken = jwsInput.readJsonContent(AccessToken.class);
+        } catch (IOException e) {
+            throw new VerificationException("Couldn't parse token signature", e);
+        }
+
+        URL backendUrl = new URL(accessToken.getIssuer());
+        URL baseUrlToCall = new URL(baseUrl);
+        if (!backendUrl.getHost().equalsIgnoreCase(baseUrlToCall.getHost())) {
+            if (hostSynonyms.contains(backendUrl.getHost())) {
```

Comment:
I don't believe adding more and more command line options for "default" cases is the best way to go forward. Canonical host names when bound to * or a specific interface should be handled by the code. Command line options should give overrides for special cases.